assets/js/components/settings/SettingsModule.test.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * SettingsModule component tests.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { createHashHistory } from 'history';
+import { Switch, Route } from 'react-router-dom';
+
+/**
+ * Internal dependencies
+ */
+import SettingsModule from './SettingsModule';
+import { render, fireEvent, createTestRegistry, provideModules } from '../../../../tests/js/test-utils';
+
+describe( 'SettingsModule', () => {
+	const mockAnalyticsModule = {
+		active: true,
+		autoActivate: true,
+		dependantModulesText: '',
+		description: 'Analytics test module description',
+		homepage: '#',
+		name: 'Analytics',
+		provides: [ 'foo', 'bar' ],
+		setupComplete: false,
+		slug: 'analytics',
+	};
+
+	const SettingsModuleWithWrapper = () => (
+		<Switch>
+			<Route path={ [ '/connected-services/:moduleSlug/:action', '/connected-services/:moduleSlug', '/connected-services' ] }>
+				<SettingsModule module={ mockAnalyticsModule } />
+			</Route>
+		</Switch>
+	);
+
+	// Create hash history to interact with HashRouter using `history.push`
+	const history = createHashHistory( { initialEntries: [ '/' ] } );
+	let registry;
+
+	beforeEach( () => {
+		global.location.hash = '';
```

Comment:
I wonder if we should be resetting this after a test or even restoring it? But resetting it at least might be a good idea.